video/out/w32_common.c

```diff
@@ -1713,6 +1740,26 @@ static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam,
     case WM_SHOWMENU:
         mp_win32_menu_show(w32->menu_ctx, w32->window);
         break;
+    case WM_TOUCH: {
+        UINT count = LOWORD(wParam);
+        TOUCHINPUT *inputs = talloc_array_ptrtype(NULL, inputs, count);
+        if (GetTouchInputInfo((HTOUCHINPUT)lParam, count, inputs, sizeof(TOUCHINPUT))) {
+            for (UINT i = 0; i < count; i++) {
+                TOUCHINPUT ti = inputs[i];
+                POINT pt = {TOUCH_COORD_TO_PIXEL(ti.x), TOUCH_COORD_TO_PIXEL(ti.y)};
+                ScreenToClient(w32->window, &pt);
+                if (ti.dwFlags & TOUCHEVENTF_DOWN)
+                    mp_input_add_touch_point(w32->input_ctx, ti.dwID, pt.x, pt.y);
+                else if (ti.dwFlags & TOUCHEVENTF_MOVE)
+                    mp_input_update_touch_point(w32->input_ctx, ti.dwID, pt.x, pt.y);
+                else if (ti.dwFlags & TOUCHEVENTF_UP)
+                    mp_input_remove_touch_point(w32->input_ctx, ti.dwID);
```

Comment:
[docs](https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-touchinput#remarks) suggest that MOVE and UP can appear simultaneously